cmake/SetupCompilerOptions.cmake

```diff
@@ -333,21 +333,42 @@ blt_append_custom_compiler_flag(
      )
 
 #
-# Modify flags to avoid static linking runtime issues on windows.
-# (adapted from RAJA)
+# Modify flags to avoid static linking runtime issues on MS Windows.
+#
+# When building on Windows, you can link in the runtime library
+#   - statically (with /MT), or
+#   - dynamically (with /MD).
+# See https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=vs-2019.
+#
+# Mixing /MD with /MT can cause linking errors.  CMake specifies
+# /MD when generating project files for MSVC and provides no way to
+# change this.  This can be a problem with Google Test in particular,
+# which when building statically replaces all /MD with /MT.  HDF5, on
+# the other hand, sternly warns against the use of /MT, even when
+# built statically.
+#
+# See https://gitlab.kitware.com/cmake/community/wikis/FAQ#dynamic-replace.
```

Comment:
Thanks @agcapps -- this is interesting.

After this PR, we can consider directly incorporating this when the user's cmake is at ``3.15+``

Reply:
great note explaining this!